src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Validation/ValidationVisitor.cs

```diff
@@ -192,6 +196,27 @@ protected virtual bool Visit(ModelMetadata metadata, string key, object model)
                 return true;
             }
 
+            if (++ValidationDepth > MaxValidationDepth)
```

Comment:
@dougbu here's the relevant source change. The tests can be updated once the design is fleshed out. 
Note that this is a Design PR.

Reply:
Sorry, missed that this was a design PR&hellip;

Reply:
This calculation excludes `MaxValidationDepth` from the depths validation can reach. `ValidationDepth++ > MaxValidationDepth` or `++ValidationDepth >= MaxValidationDepth` would be much more intuitive. E.g. `MaxValidationDepth = 1` means the visitor will happily visit the first (top) level in your model(s).